### PR TITLE
DINT-68-Provide a UI for the Reference Application 

### PR DESCRIPTION
**Pull request by ssmusoke**
1. Refactored classes and packages to match the layout of other OpenMRS modules
2. Renamed RuleDefn and RuleDefnLoader to RuleDefinition and RuleDefinitionLoader 
3. Standardized all class names to camel case 
4. Added the ability to run a single rule 
5. Added the ability for the rules to provide metadata to save rules to the database instead of a liquibase changeset if desired 
6. Added a data quality link to the Home page to show the list of results 
7. Added GSP pages to:
8. Show the results of running all the rules
9. Show the available rules as well as run a single rule
10. Removed explicit transaction management replacing it with @Transactional annotation 

A demonstration can be found at http://bit.ly/ugandaemr-demo with credentials admin/test

### CODE REVIEW

**Comment by dkayiwa** [comment]
@ssmusoke this pull request does not look right. Did you notice it too? :)

**Comment by ssmusoke** [comment]
@dkayiwa Thanks for catching that, squashed all commits into a single one

**Comment by ssmusoke** [comment]
@bharatak @deepaucksharma FYA

File: README.md [hunk 1]
```diff
@@ -1,6 +1,69 @@
...
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.dataintegrity.rule.RuleDefinition;
+import org.openmrs.module.dataintegrity.rule.RuleResult;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class InvalidDateOfDeath implements RuleDefn<Patient> {
```
Comment (bharatak) [inline README.md]: Need to change the interface to RuleDefinition

Comment (ssmusoke) [reply README.md]: @bharatak had forgotten that, will update the whole README file

**Comment by dkayiwa** [comment]
@ssmusoke did you notice the merge conflicts in this pull request?

**Comment by ssmusoke** [comment]
@dkayiwa the new file overwrites the existing ones

**Comment by dkayiwa** [comment]
@ssmusoke this pull request is in a state where it cannot be merged as is. You need to look into it.

**Comment by ssmusoke** [comment]
@dkayiwa will work on resolving the merge conflicts ASAP

**Comment by dkayiwa** [comment]
@ssmusoke i still see multiple commits in this. :)

**Comment by dkayiwa** [comment]
@ssmusoke did you give up on this? :)

**Comment by ssmusoke** [comment]
@dkayiwa I am finalizing the conflict resolution first thing tomorrow morning

**Comment by ssmusoke** [comment]
@bharatak @deepaucksharma Please review this PR

@dkayiwa I have left two commits, because I am unable to get the conflicts resolved and the changes rebased in a single one (5 failures so far) :-(

**Comment by dkayiwa** [comment]
@ssmusoke very big pull requests like this, which combine lots of things, are hard to review. If i were you, i would create multiple pull requests where each is doing something specific. :)

**Comment by ssmusoke** [comment]
@dkayiwa You make a good point @bharatak @deepaucksharma - let me do just that

**Pull request closed without merging**